trade-finance-manager-api/src/v1/controllers/deal.tasks.js

```diff
@@ -4,18 +4,31 @@ const { createTasks } = require('../helpers/create-tasks');
 
 /**
  * Check if the "create or match parties" task should be created
+ * if BSS, show task if either buyer or exporter party URN are not present
+ * if GEF, show task if exporter party URN is not present
  * @param {Object} deal
  * @returns {Boolean}
  */
 const shouldCreatePartiesTask = (deal) => {
-  const { tfm } = deal;
-  const exporterPartyUrn = tfm.parties.exporter.partyUrn;
-
-  if (exporterPartyUrn && exporterPartyUrn.length) {
-    return false;
+  const { dealType, tfm } = deal;
```

Comment:
Need a `trim` function to ensure input does not have any white spaces both in leading and trailing area.